docs/src/compatibility.md

```diff
@@ -56,6 +56,9 @@ Example = "0.2.1"
 
 results in a versionbound on Example as `[0.2.1, 0.3.0)`.
 
+In particular, it is valid to register a package with version 0.2.4 when it has feature additions compared to 0.2.3 as
```

Comment:
We have not really used the term `register` in this text, since this is talking about what Pkg considers compatible. Maybe rewrite it without that?

Reply:
Thanks for the feedback. I reworded it in the new commit.